Signal/src/view controllers/CallInterstitialViewController.swift

```diff
@@ -0,0 +1,166 @@
+
+        // Dark blurred background.
+        let blurEffect = UIBlurEffect(style: .dark)
+        blurView = UIVisualEffectView(effect: blurEffect)
+        blurView.isUserInteractionEnabled = false
+        self.view.addSubview(blurView)
+
+        contentView = UIView()
+        self.view.addSubview(contentView)
+
+        let dialingLabel = UILabel()
+        dialingLabel.text = NSLocalizedString("CALL_INTERSTITIAL_CALLING_LABEL", comment: "Title for call interstitial view")
+        dialingLabel.textColor = UIColor.white
+        dialingLabel.font = UIFont.ows_lightFont(withSize:ScaleFromIPhone5To7Plus(32, 40))
+        dialingLabel.textAlignment = .center
+        contentView.addSubview(dialingLabel)
+
+        let cancelCallButton = UIButton()
+        cancelCallButton.setTitle(NSLocalizedString("CALL_INTERSTITIAL_CANCEL_BUTTON", comment: "Label for cancel button on call interstitial view"),
+                                  for:.normal)
+        cancelCallButton.setTitleColor(UIColor.white, for:.normal)
+        cancelCallButton.titleLabel?.font = UIFont.ows_lightFont(withSize:ScaleFromIPhone5To7Plus(26, 32))
+        let buttonInset = ScaleFromIPhone5To7Plus(7, 9)
+        cancelCallButton.titleEdgeInsets = UIEdgeInsets(top: buttonInset,
+                                                        left: buttonInset,
+                                                        bottom: buttonInset,
+                                                        right: buttonInset)
+        cancelCallButton.addTarget(self, action:#selector(cancelCallButtonPressed), for:.touchUpInside)
+        contentView.addSubview(cancelCallButton)
+
+        dialingLabel.autoPinWidthToSuperview()
+        dialingLabel.autoVCenterInSuperview()
+
+        cancelCallButton.autoSetDimension(.height, toSize:ScaleFromIPhone5To7Plus(50, 60))
+        cancelCallButton.autoPinWidthToSuperview()
+        cancelCallButton.autoPinEdge(toSuperviewEdge:.bottom, withInset:ScaleFromIPhone5To7Plus(23, 41))
+    }
+
+    func cancelCallButtonPressed(sender button: UIButton) {
+        cancelCall()
+    }
+
+    // MARK: - Layout
+
+    override func updateViewConstraints() {
+        if !hasConstraints {
+            // We only want to create our constraints once.
+            //
+            // Note that constraints are also created elsewhere.
+            // This only creates the constraints for the top-level contents of the view.
+            hasConstraints = true
+
+            // Force creation of the view.
+            let view = self.view
+            assert(view != nil)
+
+            // Dark blurred background.
+            blurView.autoPinEdgesToSuperviewEdges()
+
+            contentView.autoPinEdgesToSuperviewEdges()
+        }
+
+        super.updateViewConstraints()
+    }
+
+    // MARK: - Methods
+
+    func cancelCall() {
+        guard !wasCallCancelled else {
+            return
+        }
+        wasCallCancelled = true
+
+        assert(callToken != nil)
```

Comment:
Eh, I'm generally adverse to (doing the idiomatic thing and) using constructors to set these kinds of required properties, because obj-c and swift constructors get so lengthy and awkward so quickly.

But, since you prodded me, I've done it here.